### PR TITLE
setElementModel memory leak fix

### PR DESCRIPTION
**Pull request by tederis**
Fixes the memory leak described in #2672. It also removes a functionality related to previous attempts to handle geometry references.

### CODE REVIEW

File: Client/game_sa/CPedSA.cpp [hunk 1]
```diff
@@ -98,9 +98,18 @@ void CPedSA::Init()
 
 void CPedSA::SetModelIndex(DWORD dwModelIndex)
 {
-    DWORD dwFunction = FUNC_SetModelIndex;
+    // Delete any existing RwObject first
+    DWORD dwFunction = GetPedInterface()->vtbl->DeleteRwObject;
```
Comment (TheNormalnij) [inline Client/game_sa/CPedSA.cpp]: This method is declared for 'CEntitySAInterface' already

Comment (tederis) [reply Client/game_sa/CPedSA.cpp]: Do you mean DeleteRwObject?

Comment (TheNormalnij) [reply Client/game_sa/CPedSA.cpp]: Yes

Comment (tederis) [reply Client/game_sa/CPedSA.cpp]: Okay. I've made the corrections.

**Review by TheNormalnij — APPROVED** [review]

**Review by TracerDS — APPROVED** [review]

**Pull request merged**